requests-for-comments/2022-06-10-nail-exports.md

```diff
@@ -0,0 +1,114 @@
+# Request for comments: Nail Exports - Ben White
+
+## Problem statement
+
+More context here https://github.com/PostHog/product-internal/pull/308/
+
+
+### What can be Exported
+
+* Insights
+  - Visualisation - web, image
+  - Data - API, csv
+* Dashboards
+  - Visualisation - web, image (or collection of images)
+  - Data - API, csv
+* Events
+  - the actual underlying data 
+* Session Recordings
+* Persons, Cohorts, Groups
+
+### How can they be Exported
+* Embeds 
+  - Via an iframe or directly linked to a dedicated webpage for viewing the element 
+* Images
+  - Server-side rendering of the shared webpage (using chromedriver) to an image (.png)
+  - Possibly needs multiple dimension options
+* Subscriptions - email:
+  - User configured regular reports via email including exported image of the relevant item
+* Subscriptions - slack:
+  - Slack notification including the exported image(s)
+* CSV / Spreadsheet:
+  - Relevant data exported to a CSV / Spreadsheet
+* Custom Components (React)
```

Comment:
How is this different from Embeds?

Reply:
The Custom react components? From an implementation perspective completely different - embeds are an easy-ish one-liner iframe, react (or vue, svelte etc.) components means creating a separate npm module, custom code etc.